source/extensions/filters/network/sni_cluster/sni_cluster.cc

```diff
@@ -0,0 +1,31 @@
+#include "extensions/filters/network/sni_cluster/sni_cluster.h"
+
+#include "envoy/network/connection.h"
+
+#include "common/common/assert.h"
+#include "common/tcp_proxy/tcp_proxy.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace SniCluster {
+
+Network::FilterStatus SniClusterFilter::onNewConnection() {
+  absl::string_view sni = read_callbacks_->connection().requestedServerName();
+  ENVOY_CONN_LOG(trace, "sni_cluster: new connection with server name {}",
+                 read_callbacks_->connection(), sni);
+
+  if (!sni.empty()) {
+    // Set the tcp_proxy cluster to the same value as SNI
+    read_callbacks_->connection().perConnectionState().setData(
+        Envoy::TcpProxy::PerConnectionTcpProxyConfig::CLUSTER_KEY,
+        std::make_unique<Envoy::TcpProxy::PerConnectionTcpProxyConfig>(sni));
```

Comment:
That would be enforcing a universal policy on all clusters. Istio heavily uses cluster names in fqdn format. The underscore thing is going to be a painful breaking change for us.

Reply:
But if you leave this as-is, it makes it unusable for other people who want to be able to use stats right now.

Although, it's probably fine to leave it as-is for now, and if anyone comes along later and wants the other behavior, they can add a config option to do a transform.

Reply:
So this is the cluster name being passed. If people have cluster names that have underscores instead of dots, then the SNI values they pass would also be the same (except they can't use the SNI regex match). As such, even Istio can't do the regex match because our cluster name is like `outbound|80|v1|bar.svc.cluster.local` .. It doesn't matter to us in this use case because we don't use the SNI match.